tasks/update-cnb-dependency/main.go

```diff
@@ -49,57 +50,50 @@ func updateCNBDependencies() error {
 		return errors.Wrap(err, "failed to load task resources")
 	}
 
-	deprecationDate, err := NewDependencyDeprecationDate(flags.deprecationDate, flags.deprecationLink, config.Dep.ID, flags.versionLine, flags.deprecationMatch)
-	if err != nil {
-		return errors.Wrap(err, "failed to create deprecation date")
-	}
-
+	// Load, update and save dependencies
 	depsToAdd, err := loadDependenciesFromBinaryBuilds(flags.binaryBuildsPath, config.Dep, config.Orchestrator)
 	if err != nil {
 		return errors.Wrap(err, "failed to construct list of dependencies to add")
 	}
 
-	var deps Dependencies
-	err = mapstructure.Decode(config.BuildpackTOML.Metadata[DependenciesKey], &deps)
+	deps, err := config.BuildpackTOML.Dependencies()
 	if err != nil {
 		return errors.Wrap(err, "failed to decode dependencies")
 	}
-	originalDeps := deps.ExpandByStack()
-	updatedDeps, err := originalDeps.MergeWith(depsToAdd)
+
+	updatedDeps, err := deps.Update(flags, config.Dep, depsToAdd)
 	if err != nil {
-		return errors.Wrap(err, "failed to merge dependency lists")
+		return errors.Wrap(err, "failed to add new dependencies to the dependencies list")
 	}
+	config.BuildpackTOML.SaveDependencies(updatedDeps)
 
-	updatedDeps, err = updatedDeps.RemoveOldDeps(config.Dep.ID, flags.versionLine, flags.versionsToKeep)
+	// Load and update Deprecation Dates
+	deprecationDate, err := NewDependencyDeprecationDate(flags.deprecationDate, flags.deprecationLink, config.Dep.ID, flags.versionLine, flags.deprecationMatch)
 	if err != nil {
-		return errors.Wrap(err, "failed to remove old dependencies")
+		return errors.Wrap(err, "failed to create deprecation date")
 	}
 
-	updatedOrder := config.BuildpackTOML.Orders.UpdateOrderDependencyVersion(config.Dep)
-
-	var deprecationDates DeprecationDates
-	err = mapstructure.Decode(config.BuildpackTOML.Metadata[DeprecationDatesKey], &deprecationDates)
+	deprecationDates, err := config.BuildpackTOML.DeprecationDates()
 	if err != nil {
 		return errors.Wrap(err, "failed to decode deprecation dates")
 	}
+
 	updatedDeprecationDates, err := deprecationDates.Update(deprecationDate)
 	if err != nil {
 		return errors.Wrap(err, "failed to update deprecation dates")
 	}
+	config.BuildpackTOML.SaveDeprecationDates(updatedDeprecationDates)
```

Comment:
Same as above (a single method on `BuildpackTOML` to update deprecation dates instead of loading and saving).

Reply:
^^ 👍